### PR TITLE
[experimental] Create basic account

### PR DESCRIPTION
**Pull request by dmitri-perelman**
Add a function for creating basic account with no roles and no previous conditions.
For the account generation event, add a new role type being "NO_ROLE_ID".

Also added a simple demo-cli to verify the e2e process with running a local node and testing it.

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

**Comment by dmitri-perelman** [comment]
Ouch, since I changed the symlinks to the actual files, it's kind of confusing. Let me add the first commit with just changing the symlinks to the files.

**Comment by dmitri-perelman** [comment]
Split to 2 commits, the first one is just about replacing the symlink with the copy of the files.

**Comment by hariria** [comment]
@dmitri-perelman would it be possible to run `cargo x clippy` on your local, it seems to be failing ci-test / lint
```bash
error: this expression borrows a reference (`&diem_sdk::client::BlockingClient`) that is immediately dereferenced by the compiler
  --> diem-move/diem-framework/experimental/demo-cli/src/main.rs:92:10
   |
92 |     send(&client, txn)?;
   |          ^^^^^^^ help: change this to: `client`
```

**Comment by dmitri-perelman** [comment]
Sorry about the noise with all the fmt, clipper, lint etc. :). Will be more careful!

File: diem-move/diem-framework/experimental/releases/artifacts/current/build/DiemExperimental/sources/Roles.move [hunk 1]
```diff
@@ -192,9 +193,12 @@ module DiemFramework::Roles {
...
     fun has_role(account: &signer, role_id: u64): bool acquires RoleId {
-       let addr = Signer::address_of(account);
-       exists<RoleId>(addr)
-           && borrow_global<RoleId>(addr).role_id == role_id
+        let addr = Signer::address_of(account);
+        if (!exists<RoleId>(addr)) {
+            role_id == NO_ROLE_ID
+        } else {
+            borrow_global<RoleId>(addr).role_id == role_id
+        }
```
Comment (gregnazario) [inline diem-move/diem-framework/experimental/releases/artifacts/current/build/DiemExperimental/sources/Roles.move:201]: This seems overly complicated.  Why isn't it just `role_id == get_role_id(addr)`?

File: diem-move/diem-framework/experimental/releases/artifacts/current/build/DiemExperimental/sources/Roles.move [hunk 2]
```diff
@@ -226,8 +230,11 @@ module DiemFramework::Roles {
     }
 
     public fun get_role_id(a: address): u64 acquires RoleId {
-        assert!(exists<RoleId>(a), Errors::not_published(EROLE_ID));
-        borrow_global<RoleId>(a).role_id
+        if (exists<RoleId>(a)) {
+            borrow_global<RoleId>(a).role_id
```
Comment (gregnazario) [inline diem-move/diem-framework/experimental/releases/artifacts/current/build/DiemExperimental/sources/Roles.move:234]: Honestly, I'd prefer if we don't use single letter variables.  They don't read well & make it harder to understand.

File: diem-move/diem-framework/experimental/demo-cli/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+use move_core_types::{
+    ident_str,
+    language_storage::{ModuleId, TypeTag},
+};
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(about = "Demo for trove hackathon")]
+pub struct TroveHackathonDemo {
```
Comment (gregnazario) [inline diem-move/diem-framework/experimental/demo-cli/src/main.rs:26]: Did we want to rename these?

**Review by gregnazario — APPROVED** [review]

**Pull request merged**